system_metrics_collector/test/moving_average_statistics/test_moving_average_statistics.cpp

```diff
@@ -0,0 +1,194 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <thread>
+#include <atomic>
+#include "moving_average_statistics/moving_average.hpp"
+
+
+/**
+ * Test fixture
+ */
+class MovingAverageStatisticsTestFixture : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+    moving_average_statistics = std::make_shared<MovingAverageStatistics<double>>();
+  }
+
+  void TearDown() override
+  {
+    moving_average_statistics->reset();
+  }
+
+protected:
+  std::shared_ptr<MovingAverageStatistics<double>> moving_average_statistics = nullptr;
+  int expected_count = 0;
+};
+
+// Useful testing constants
+const double TEST_DATA[] = {-3.5, -2.1, -1.1, 0.0, 4.7, 5.0, 6.7, 9.9, 11.0};
+const double EXPECTED_AVG = 3.4;
+const double EXPECTED_MIN = -3.5;
+const double EXPECTED_MAX = 11.0;
+const double EXPECTED_STD = 4.997999599839919955173;
```

Comment:
Can these constants be `constexpr`?

Would it be possible to calculate `EXPECTED_AVG`, `EXPECTED_MIN`, `EXPECTED_MAX`, `EXPECTED_STD` from `TEST_DATA` using a constexpr expression?

Reply:
`std::array` has constexpr functions and may help in my second comment.

Reply:
I'm having some issues with this, maybe we can sit down or punt to another PR.

Reply:
ok